controllers/addon_controller.go

```diff
@@ -173,71 +175,77 @@ func (r *AddonReconciler) execAddon(ctx context.Context, req reconcile.Request,
 	return ret, procErr
 }
 
-// SetupWithManager is called to setup manager and watchers
-func (r *AddonReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	log := r.Log
-	managedNS := "addon-manager-system"
-
-	nsInformers := dynamicinformer.NewFilteredDynamicSharedInformerFactory(r.dynClient, time.Minute*30, managedNS, nil)
-	wfInf := nsInformers.ForResource(common.WorkflowGVR())
-	bldr := ctrl.NewControllerManagedBy(mgr).
-		For(&addonmgrv1alpha1.Addon{}).
-		// Watch workflows created by addon only in addon-manager-system namespace
-		Watches(&source.Informer{Informer: wfInf.Informer().(cache.Informer)}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &addonmgrv1alpha1.Addon{},
-		})
-
-	generatedInformers = informers.NewSharedInformerFactory(r.generatedClient, time.Minute*30)
-
-	err := mgr.Add(manager.RunnableFunc(func(s <-chan struct{}) error {
-		generatedInformers.Start(s)
-		generatedInformers.WaitForCacheSync(s)
-		nsInformers.Start(s)
-		nsInformers.WaitForCacheSync(s)
-		<-s
-		return nil
-	}))
+func New(mgr manager.Manager) (controller.Controller, error) {
+	r := &AddonReconciler{
+		Client:          mgr.GetClient(),
+		Log:             ctrl.Log.WithName(controllerName),
+		Scheme:          mgr.GetScheme(),
+		versionCache:    addon.NewAddonVersionCacheClient(),
+		dynClient:       dynamic.NewForConfigOrDie(mgr.GetConfig()),
+		generatedClient: kubernetes.NewForConfigOrDie(mgr.GetConfig()),
+		recorder:        mgr.GetEventRecorderFor("addons"),
+		statusWGMap:     map[string]*sync.WaitGroup{},
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
 	if err != nil {
-		log.Error(err, "Error adding InformerFactory to the Manager")
-		return err
+		return nil, err
+	}
+
+	// Watch workflows created by addon only in addon-manager-system namespace
+	if err := c.Watch(&source.Kind{Type: &wfv1.Workflow{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &addonmgrv1alpha1.Addon{},
+	}); err != nil {
+		return nil, err
 	}
 
 	// Watch for changes to kubernetes Resources matching addon labels.
-	for _, resc := range resources {
-		gvk := resc.GetObjectKind().GroupVersionKind()
-		_, kind := gvk.ToAPIVersionAndKind()
+	if err := c.Watch(&source.Kind{Type: &addonmgrv1alpha1.Addon{}}, &handler.EnqueueRequestForObject{}); err != nil {
```

Comment:
Because we want leverage the built-in library directly, do not need extra informers, which has already with the k8s client.

Reply:
Since you made this change why do we need `resources` array now?
https://github.com/keikoproj/addon-manager/blob/bf0af606319b0ad1489813d2283d32333996af4b/controllers/addon_controller.go#L63-L70

Reply:
Also you're using empty value for the types, which has null values.
```
&appsv1.Deployment{}
```
as opposed to resources list version is same but has kind and version populated:
```
&appsv1.Deployment{TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"}}
```